src/components/Footer.tsx

```diff
@@ -0,0 +1,92 @@
+import cn from 'classnames';
+import { FilterStatus, Todo } from '../types/Todo';
+import { deleteTodos } from '../api/todos';
+import { showError } from '../helpers/helpers';
+
+interface Props {
+  filter: string,
+  todos: Todo[]
+  setFilter: (text: FilterStatus) => void;
+  setTodos: (todos: Todo[]) => void;
+  setError: React.Dispatch<React.SetStateAction<string>>
+  setLoader: React.Dispatch<React.SetStateAction<number[]>>;
+}
+
+export const Footer: React.FC<Props> = ({
+  filter,
+  todos,
+  setFilter,
+  setTodos,
+  setError,
+  setLoader,
+}) => {
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+  const clearButtonIsVisible = todos.some(todo => todo.completed);
+  const clickHandler = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+    const activeTodos = todos.filter(todo => !todo.completed);
+
+    completedTodos.map(todo => {
+      setLoader(prevState => [...prevState, todo.id]);
+
+      return deleteTodos(todo.id);
+    });
+    try {
+      await Promise.all(completedTodos);
+      setTodos(activeTodos);
+      setLoader([]);
+    } catch (error) {
+      showError('Unable to delete a todo', setError);
+      setLoader([]);
+    }
+  };
+
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {`${itemsLeft} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={cn('filter__link', {
+            selected: filter === FilterStatus.ALL,
+          })}
+          onClick={() => setFilter(FilterStatus.ALL)}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={cn('filter__link', {
+            selected: filter === FilterStatus.ACTIVE,
+          })}
+          onClick={() => setFilter(FilterStatus.ACTIVE)}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
+          className={cn('filter__link', {
+            selected: filter === FilterStatus.COMPLETED,
+          })}
+          onClick={() => setFilter(FilterStatus.COMPLETED)}
+        >
+          Completed
+        </a>
```

Comment:
you can iterate through your enum, it will definitely improve your code, since you'll only have to add new property to enum, and it will immediately pop up on the page

Reply:
we have  trouble with link  for All because in this case our link #/, of course we can create object with text/link and use here instead of enum but i think in our app this aproach it also ok